app/services/testing_ground/save_as.rb

```diff
@@ -18,14 +18,16 @@ def initialize(original, name, user = nil)
 
   def run
     duplicate = @original.dup
+    duplicate.name = @name
+    duplicate.user = @user
 
-    TestingGround.transaction do
-      duplicate.name = @name
-      duplicate.user = @user
+    if duplicate.valid?
```

Comment:
> and catch it in the controller like so:

This was my first approach. But I did it like this:

````ruby
def save_as
  begin
    @testing_ground = TestingGround::SaveAs.run(
      @testing_ground, testing_ground_params[:name], current_user
    )

  render json: { redirect: testing_ground_path(@testing_ground) }
rescue ActiveRecord::RecordInvalid
  render json: { errors: @testing_ground.errors }, status: 422
end
````

However this caused the `@testing_ground` to point to the old testing ground and not towards the duplicate. Maybe `catch` prevents this behavior though.